query/src/test/java/com/connexta/replication/query/cmp/QueryServiceComponentTest.java

```diff
@@ -0,0 +1,421 @@
+  private static final MeterRegistry REGISTRY = new MeterRegistryMock();
+
+  private final QueueBroker broker =
+      new MemoryQueueBroker(
+          QueryServiceComponentTest.QUEUE_CAPACITY, QueryServiceComponentTest.REGISTRY);
+
+  @Autowired private FilterIndexManager indexManager;
+  @Autowired private FilterManager filterManager;
+  @Autowired private SiteManager siteManager;
+
+  @Rule public final MultiThreadedErrorCollector collector = new MultiThreadedErrorCollector();
+
+  private NodeAdapters nodeAdapters = mock(NodeAdapters.class);
+  private NodeAdapterFactory adapterFactory = mock(NodeAdapterFactory.class);
+  private final Set<Task> tasks = new HashSet<>();
+  private final NodeAdapter adapter = mock(NodeAdapter.class);
+
+  @Before
+  public void setup() {
+    when(adapterFactory.create(any(URL.class))).thenReturn(adapter);
+    when(nodeAdapters.factoryFor(SiteType.DDF)).thenReturn(adapterFactory);
+    when(nodeAdapters.factoryFor(SiteType.ION)).thenAnswer(collectAdapterException());
+    when(adapter.isAvailable()).thenReturn(true);
+  }
+
+  private Answer<?> collectAdapterException() {
+    return (Answer<Object>)
+        invocationOnMock -> {
+          Throwable t = new AdapterException("no adapters for ION sites should be retrieved");
+          collector.addError(t);
+          throw t;
+        };
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    tasks.clear();
+    filterManager.objects().map(Filter::getId).forEach(filterManager::remove);
+    siteManager.objects().map(Site::getId).forEach(siteManager::remove);
+  }
+
+  /*
+   Verifies that multiple metacards can be handled and that the proper date is saved in the filterIndex
+  */
+  @Test
+  public void testQueryMultipleMetacards() throws Exception {
+    final Site site = new SiteImpl(generateSitePojo(1));
+    siteManager.save(site);
+    final Filter filter = new FilterImpl(generateFilterPojo("1", 11));
+    filterManager.save(filter);
+    List<Metadata> metadataList =
+        List.of(
+            generateMetadataWithResource(111),
+            generateMetadataWithResource(112),
+            generateMetadataWithResource(113, new Date(1)));
+    when(adapter.query(Mockito.argThat(isQueryRequestWithFilter(filter.getFilter()))))
+        .thenReturn(
+            new QueryResponseImpl(
+                metadataList,
+                e -> {
+                  throw new AssertionError(e);
+                }));
+
+    indexManager = Mockito.spy(indexManager);
+
+    startQueryManager(THIRTY_MINUTES);
+
+    final MemorySiteQueue queue = (MemorySiteQueue) broker.getQueue(site.getId());
+
+    assertThat(
+        "timed out waiting for expected published task",
+        queue.waitForPendingSizeToReach(3, 5L, TimeUnit.SECONDS),
+        Matchers.equalTo(true));
+
+    for (Metadata metadata : metadataList) {
+      assertTask(queue.take(), metadata, (byte) 0, OperationType.HARVEST);
+    }
+    verify(indexManager)
```

Comment:
✏️ How about we verify the index before we pull out tasks since we know te query service should have completed its pass and updated it. I would suggest we do that on all of the tests.